src/goto-programs/goto_clean_expr.cpp

```diff
@@ -423,7 +427,7 @@ void goto_convertt::clean_expr(
     // This is simply replaced by the literal
     DATA_INVARIANT(
       expr.operands().size() == 1, "ID_compound_literal has a single operand");
-    expr=expr.op0();
+    expr = to_unary_expr(expr).op();
```

Comment:
I'd remove the `DATA_INVARIANT` above. (I'm generally not happy that we redundantly invent new error messages and checks that run the risk of going out of sync. `to_..._expr` should be the authoritative place.)

Reply:
done